src/Stratis.Bitcoin.Features.SmartContracts/Rules/AllowedScriptTypeRule.cs

```diff
@@ -25,18 +24,13 @@ public override Task RunAsync(RuleContext context)
 
             foreach (Transaction transaction in block.Transactions)
             {
-                this.CheckTransaction(transaction);
+                CheckTransaction(this.network, transaction);
             }
 
             return Task.CompletedTask;
         }
 
-        public void CheckTransaction(MempoolValidationContext context)
-        {
-            this.CheckTransaction(context.Transaction);
-        }
-
-        private void CheckTransaction(Transaction transaction)
+        public static void CheckTransaction(Network network, Transaction transaction)
```

Comment:
Feel a little bit weird about this. Only better I solution I know of is to create a class that both this and the mempool rule can consume. What do you think @rowandh ?

Reply:
It's not ideal, but if it's correct I think I'd rather see this PR go in than block over it.